src/main/kotlin/uk/gov/gdx/datashare/service/DbPasswordService.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.gdx.datashare.service
+
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.rds.RdsUtilities
+import software.amazon.awssdk.services.rds.model.GenerateAuthenticationTokenRequest
+
+@Service
+class DbPasswordService(
+  @Value("\${db.provider}") val dbProvider: DbProvider,
+  @Value("\${db.hostname}") val dbHostname: String,
+  @Value("\${db.port}") val dbPort: Int,
+  @Value("\${db.username}") val dbUsername: String,
+  @Value("\${db.password:#{null}}") val dbPassword: String?
+) {
+  enum class DbProvider {
+    LOCALSTACK,
+    RDS
+  }
+
+
+  val rdsUtilities: RdsUtilities = RdsUtilities.builder()
+    .credentialsProvider(DefaultCredentialsProvider.create())
+    .region(Region.EU_WEST_2)
+    .build()
+
+
+  fun password(): String {
```

Comment:
could we add a username as well, so that we can just use the service instead of the db.username config value in the other spaces?